packages/pigeon/lib/ast.dart

```diff
@@ -82,39 +74,84 @@ class Api extends Node {
   }
 }
 
-/// Represents a field on a [Class].
-class Field extends Node {
-  /// Parametric constructor for [Field].
-  Field({
-    required this.name,
+/// An entity that represents a typed concept, like a [TypeDeclaration] or [NamedType].
+abstract class TypedEntity {
+  /// The data-type of the entity (ex 'String' or 'int').
+  String get dataType;
+
+  /// The type arguments to the entity.
+  List<TypeDeclaration>? get typeArguments;
```

Comment:
stuart is referring to the review here: https://github.com/flutter/packages/pull/416

In summary, what is still relevant:
1) Wants type declarations in the constructors for Fields
1) Naming of methods in TypedEntity is confusing, particularly the dataType field.
1) Sometimes `dataType` sometimes `type` why?
1) Do we still want TypedEntity?

Reply:
1) Done
2) I updated the docstring for the dataType field
3) Hopefully it is consistent now, "type" = TypeDeclaration, "dataType" = the base name of a TypeDeclaration.  I think "dataType" is probably as good of a name as we can get since we don't want it to be confused with `NamedType.name`.
4) We talked in person, I think it's still a fine abstraction.  I see the benefits either way.  I removed it a few times and kept reverting it since it wasn't an obvious improvement.